pkg/snapstore/abs_snapstore.go

```diff
@@ -78,69 +83,62 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 }
 
 // ConstructBlobServiceURL constructs the Blob Service URL based on the activation status of the Azurite Emulator.
-// It checks the environment variables for emulator configuration and constructs the URL accordingly.
-// The function expects two environment variables:
+// It checks the environment variable for emulator configuration and constructs the URL accordingly.
+// The function expects the following environment variable:
 // - AZURE_EMULATOR_ENABLED: Indicates whether the Azurite Emulator is enabled (expects "true" or "false").
-// - AZURE_STORAGE_API_ENDPOINT: Specifies the Azurite Emulator endpoint when the emulator is enabled.
-func ConstructBlobServiceURL(credentials *azblob.SharedKeyCredential) (*url.URL, error) {
-	defaultURL, err := url.Parse(fmt.Sprintf("https://%s.%s", credentials.AccountName(), brtypes.AzureBlobStorageHostName))
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse default service URL: %w", err)
-	}
+func ConstructBlobServiceURL(storageAccount, domain string) (*url.URL, error) {
+	scheme := "https"
+
 	emulatorEnabled, ok := os.LookupEnv(EnvAzureEmulatorEnabled)
-	if !ok {
-		return defaultURL, nil
-	}
-	isEmulator, err := strconv.ParseBool(emulatorEnabled)
-	if err != nil {
-		return nil, fmt.Errorf("invalid value for %s: %s, error: %w", EnvAzureEmulatorEnabled, emulatorEnabled, err)
-	}
-	if !isEmulator {
-		return defaultURL, nil
-	}
-	endpoint, ok := os.LookupEnv(AzuriteEndpoint)
-	if !ok {
-		return nil, fmt.Errorf("%s environment variable not set while %s is true", AzuriteEndpoint, EnvAzureEmulatorEnabled)
+	if ok {
+		isEmulator, err := strconv.ParseBool(emulatorEnabled)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value for %s: %s, error: %w", EnvAzureEmulatorEnabled, emulatorEnabled, err)
+		}
+		if isEmulator {
+			// TODO: going forward, use Azurite with HTTPS (TLS) communication
+			scheme = "http"
+		}
 	}
-	// Application protocol (http or https) is determined by the user of the Azurite, not by this function.
-	return url.Parse(fmt.Sprintf("%s/%s", endpoint, credentials.AccountName()))
+
+	return url.Parse(fmt.Sprintf("%s://%s.%s", scheme, storageAccount, domain))
 }
 
-func getCredentials(prefixString string) (string, string, error) {
+func getCredentials(prefixString string) (*absCredentials, error) {
 	if filename, isSet := os.LookupEnv(prefixString + absCredentialJSONFile); isSet {
```

Comment:
I would not like to change anything more in this PR. Maybe in a different one. We anyway need to overhaul the credential passing in etcdbr, because it's quite inconsistent right now across providers.

Reply:
We'll have to introduce new flags which specify the path at which the credential files are stored.
Perhaps we could do that in another PR since this environment variable only passes the *path* of the secret file and not the contents of a secret itself, the latter of which this PR is aiming to fix.